File: src/main/java/com/github/schlak/database/Implementation/MSSQL/FixedValues/MSSQLDataType.java
package com.github.schlak.database.Implementation.MSSQL.FixedValues;


import com.github.schlak.database.Definition.FixedValues.BasicDataType;

/**
 * Created by jonas on 15.03.17.
 */
public enum MSSQLDataType {

    BLOB("varbinary(MAX)"),
    INTEGER("integer"),
    FLOAT("float"),
    TEXT("ntext");

    private String value;

    private MSSQLDataType(String value) {
        this.value = value;
    }

    public static MSSQLDataType get(BasicDataType baseBaseDataType) {
        switch (baseBaseDataType) {
            case REAL:
                return FLOAT;
            case INTEGER:
                return INTEGER;
            case TEXT:
                return TEXT;
        }
        return BLOB;
    }

}

File: src/main/java/com/github/schlak/database/Implementation/MSSQL/FixedValues/MSSQLDataType.kt
package com.github.schlak.database.Implementation.MSSQL.FixedValues

import com.github.schlak.database.Definition.FixedValues.BasicDataType

/**
 * Created by Jonas Schlak on 21.03.2017.
 */
enum class MSSQLDataType(val value : String) {

    BLOB("varbinary(MAX)"),
    INTEGER("integer"),
    FLOAT("float"),
    TEXT("ntext");

    companion object{
        fun get(baseBaseDataType : BasicDataType): MSSQLDataType {

            when (baseBaseDataType){
                BasicDataType.REAL -> return FLOAT;
                BasicDataType.INTEGER -> return INTEGER;
                BasicDataType.TEXT -> return  TEXT;
                else -> {
                    return BLOB;
                }
            }
        }
    }
}